validator/src/commands/repair_shred_from_peer/mod.rs

```diff
@@ -37,16 +56,76 @@ pub fn command(_default_args: &DefaultArgs) -> App<'_, '_> {
 }
 
 pub fn execute(matches: &ArgMatches, ledger_path: &Path) -> Result<(), String> {
-    let pubkey = value_t!(matches, "pubkey", Pubkey).ok();
-    let slot = value_t_or_exit!(matches, "slot", u64);
-    let shred_index = value_t_or_exit!(matches, "shred", u64);
+    let RepairShredFromPeerArgs {
+        pubkey,
+        slot,
+        shred,
+    } = RepairShredFromPeerArgs::from_clap_arg_match(matches)?;
+
     let admin_client = admin_rpc_service::connect(ledger_path);
     admin_rpc_service::runtime()
         .block_on(async move {
             admin_client
                 .await?
-                .repair_shred_from_peer(pubkey, slot, shred_index)
+                .repair_shred_from_peer(pubkey, slot, shred)
                 .await
         })
         .map_err(|err| format!("repair shred from peer request failed: {err}"))
 }
+
+#[cfg(test)]
+mod tests {
+    use {super::*, crate::commands::tests::verify_args_struct_by_command, std::str::FromStr};
+
+    #[test]
+    fn verify_args_struct_by_command_repair_shred_from_peer_missing_slot() {
+        let app = command();
+        let matches = app.get_matches_from(vec![COMMAND]);
+        let args = RepairShredFromPeerArgs::from_clap_arg_match(&matches);
+        assert_eq!(args, Err("slot is not a valid number".to_string()));
```

Comment:
nit: This test is technically dependent on the fact that we parse `slot` before `shred` which is an implementation detail. I'd probably be fine with removing the error text from this assertion

Reply:
+1, the important bit is that error is reported. If we want to track specific errors, we should give errors unique IDs, and then match in that.